packages/language-model/src/fal.ts

```diff
@@ -124,3 +125,65 @@ export function getImageGenerationModelProvider(
 
 	return undefined;
 }
+
+export interface FalImage {
+	url: string;
+	width: number;
+	height: number;
+	content_type: string;
+}
+
+export interface UsageCalculator {
+	calculateUsage(images: FalImage[]): {
+		output: number;
+		unit: "IMAGES";
+	};
+}
+
+export class PixelBasedUsageCalculator implements UsageCalculator {
+	calculateUsage(images: FalImage[]) {
+		const totalPixels = images.reduce(
+			(sum, image) => sum + image.height * image.width,
```

Comment:
use fal.ai API directly to obtain size of generated image
because `generatiImage()` of AI SDK [does not return size info of generateg image at this point](https://sdk.vercel.ai/docs/reference/ai-sdk-core/generate-image#returns)